crates/torii/core/src/engine.rs

```diff
@@ -404,9 +404,12 @@ impl<P: Provider + Sync> Engine<P> {
 
     async fn process_block(&mut self, block_number: u64, block_timestamp: u64) -> Result<()> {
         for processor in &self.processors.block {
-            processor
+            if let Err(e) = processor
```

Comment:
I just want to make sure, in this case, we should retry the processor, otherwise the indexer will end up in an undefined state. We generally want to panic when we run into something that we can't handle so we can detect it and address the root issue.

Reply:
Hum, this is actually what we try to solve here. Due to the permission less nature of the world, if someone emit an invalid event by hand, torii will panic forever.

When a processor must be validated, a test suite must be done to ensure it works properly. But when we are in production, if Torii can't handle something it must not panic.

We may then have some control at the world level to ensure that an event is always emitted for a registered model. But we may have other invalid situation such keys that couldn't always be checked on chain.

Reply:
I think maybe we should have some events that could panic and others not? Or have some sort of potential fallback like we previously discussed where, if the event fails, handle the error and try to reconstruct the model schema  without the ModelRegistered event. Not sure how it would be implemented tho

Reply:
Sorry I wasn't specific on the first comment. Totally agree on the fact that transaction and block processors are not expected to fail and we should retry or even still panicking.

On the event side, we're too subject to wrong data being captured by torii, and by ignoring those torii is ensured to run even if someone does voluntarily something bad on the world emitting wrong event structure.

Reply:
@tarrencev ok forget my answer, I just didn't see correctly where your comment was pointing to the block processor specifically. My bad on that.

Reply:
Fixed in 007b7a151adcc08e59ddf954cd8ab7aa9db0483d where block and transaction processors will still panic if there is a problem with the parsing.
As we're not dependent on the network during the process, I propose to not retry and keep the panic as we had before.